src/common_modules/vsts-cli-build-common/vsts/cli/build/common/build_definition.py

```diff
@@ -36,14 +38,16 @@ def build_definition_list(name=None, top=None, team_instance=None, project=None,
                                                                             repo=repository)
     client = get_build_client(team_instance)
     query_order = 'DefinitionNameAscending'
-    repository_type = None
+
     if repository is not None:
-        resolved_repository = _resolve_repository_as_id(repository, team_instance, project)
-        if resolved_repository is None:
-            raise ValueError("Could not find a repository with name, '{}', in project, '{}'.".format(repository,
-                                                                                                        project))
-        else:
+        if repository_type is None:
             repository_type = 'TfsGit'
```

Comment:
Should not resolve_repository_as_id take repository_type property?

Reply:
That function is only for TFSGit repositoryType. As we are calling Git Client in that and fetching repositories.